bin/doctest

```diff
@@ -2,18 +2,25 @@
 
 'use strict';
 
-// Based on:
-// https://github.com/jashkenas/coffeescript/blob/1.9.1/src/command.coffee#L431-L441
-var args = process.argv.slice (1);
+var path = require ('path');
+
+var args = process.argv.slice (2);
 var idx = args.indexOf ('--nodejs');
-if (idx < 0 || idx === args.length - 1) {
-  require ('../lib/command');
-} else {
-  require ('child_process')
-  .spawn (process.execPath,
-          args[idx + 1]
-            .split (/\s+/)
-            .concat (args.slice (0, idx), args.slice (idx + 2)),
-          {cwd: process.cwd (), env: process.env, stdio: [0, 1, 2]})
-  .on ('exit', process.exit);
-}
+var esm = Number (process.versions.node.split ('.')[0]) >= 9;
+var flags = idx >= 0 && idx < args.length - 1;
+
+var nodeargs = []
+  .concat (flags ? args[idx + 1].split (/\s+/) : [])
+  .concat (esm ? ['--experimental-modules'] : []);
+
+var script = path.resolve (__dirname, '..', 'lib', 'command');
+
+var scriptargs = flags ?
+  args.slice (0, idx).concat (args.slice (idx + 2)) :
+  args;
+
+require ('child_process').spawn (
+  process.execPath,
+  nodeargs.concat ([script]).concat (scriptargs),
```

Comment:
I would prefer to see one big concatenation:

```javascript
[].concat (flags ? args[idx + 1].split (/\s+/) : [])
  .concat (esm ? ['--experimental-modules'] : [])
  .concat ([path.resolve (__dirname, '..', 'lib', 'command')])
  .concat (flags ? (args.slice (0, idx)).concat (args.slice (idx + 2)) : args)
```

Reply:
The last line doesn't fit, so is broken. Comparing the script before and after, I think it looks more messy after, but I'll leave the decision to you.

Reply:
For me it's less about how the code looks than about how it reads. I found mentally evaluating `nodeargs.concat ([script]).concat (scriptargs)` difficult for two reasons:

  - I had to look up three variables; and
  - I had to look for dependencies between the variables.